pkg/operator/staticpod/controller/node/node_controller.go

```diff
@@ -106,14 +105,33 @@ func (c NodeController) sync() error {
 		newTargetNodeStates = append(newTargetNodeStates, operatorv1.NodeStatus{NodeName: node.Name})
 	}
 
-	operatorStatus.NodeStatuses = newTargetNodeStates
-	if !equality.Semantic.DeepEqual(originalOperatorStatus, operatorStatus) {
-		if _, updateError := c.operatorClient.UpdateStaticPodOperatorStatus(resourceVersion, operatorStatus); updateError != nil {
-			return updateError
+	// detect and report master nodes that are not ready
+	notReadyNodes := []string{}
+	for _, node := range nodes {
+		for _, con := range node.Status.Conditions {
+			if con.Type == coreapiv1.NodeReady && con.Status != coreapiv1.ConditionTrue {
+				notReadyNodes = append(notReadyNodes, node.Name)
+				c.eventRecorder.Eventf("MasterNodeNotReady", "the master node %s is not ready", node.Name)
+			}
 		}
 	}
-
-	return nil
+	cond := operatorv1.OperatorCondition{
+		Type:   condition.NodeControllerDegradedConditionType,
+	}
+	if len(notReadyNodes) > 0 {
+		cond.Status = operatorv1.ConditionTrue
+		cond.Reason = "NotReady"
+	    cond.Message = fmt.Sprintf("The master node(s) %v not ready", notReadyNodes)
+	} else {
+	    cond.Status = operatorv1.ConditionFalse
+		cond.Reason = "Ready"
+		cond.Message = "All master nodes are ready"
+	}
+	_, _, updateError := v1helpers.UpdateStaticPodStatus(c.operatorClient, v1helpers.UpdateStaticPodConditionFn(cond), func(status *operatorv1.StaticPodOperatorStatus) error{
+		status.NodeStatuses = newTargetNodeStates
```

Comment:
alternatively, we could extend `[]NodeStatus` and change the other controllers to respect it.